lib/ClangImporter/ClangModuleDependencyScanner.cpp

```diff
@@ -28,11 +28,11 @@ using namespace clang::tooling;
 using namespace clang::tooling::dependencies;
 
 class swift::ClangModuleDependenciesCacheImpl {
-  /// The name of the file used for the "import hack" to compute module
+  /// Cache the names of the files used for the "import hack" to compute module
   /// dependencies.
   /// FIXME: This should go away once Clang's dependency scanning library
   /// can scan by module name.
-  std::string importHackFile;
+  llvm::DenseMap<StringRef, std::string> importHackFileCache;
```

Comment:
Please use `StringMap` here, because I don't think it's guaranteed that the keys will live forever (and they won't be that large overall).